wire-ios-sync-engine/Source/Synchronization/SyncAgent.swift

```diff
@@ -104,9 +104,9 @@ final class SyncAgent: NSObject, SyncAgentProtocol {
     /// This method logs any errors and does not wait for the sync to finish.
 
     func resume() {
-        Task {
+        ongoingSyncTask = Task {
```

Comment:
This task will complete after we've processed the pending events. What would happen if we cancel it when it's already finished? Are you sure that the live sync task spawned inside will be cancelled too?